src/networks.jl

```diff
@@ -271,21 +274,232 @@ function (net::NoiseConditionalScoreNetwork)(x, c, t)
+        tconv1=Conv(Tuple(proj_kernelsize for _ in 1:nspatial), channels[1] + channels[1] => outchannels, stride=1, pad=SamePad())
+    end
+    
+    # Bulk of the network
+    layers = (
+        gaussfourierproj=GaussianFourierProjection(embed_dim, scale),
+        linear=Dense(embed_dim, embed_dim, swish),
+        
+        # Lifting
+        conv1=conv1,
+        dense1=Dense(embed_dim, channels[1]),
+        gnorm1=GroupNorm(channels[1], 4, swish),
+        
+        # Encoding
+        conv2=Downsampling(channels[1] => channels[2], nspatial, kernel_size=3, periodic=periodic),
+        dense2=Dense(embed_dim, channels[2]),
+        gnorm2=GroupNorm(channels[2], 32, swish),
+        
+        conv3=Downsampling(channels[2] => channels[3], nspatial, kernel_size=3, periodic=periodic),
+        dense3=Dense(embed_dim, channels[3]),
+        gnorm3=GroupNorm(channels[3], 32, swish),
+        
+        conv4=Downsampling(channels[3] => channels[4], nspatial, kernel_size=3, periodic=periodic),
+        dense4=Dense(embed_dim, channels[4]),
+        
+        # Residual Blocks
+        resnet_blocks = 
+        [ResnetBlockNCSN(channels[end], nspatial, embed_dim; p = dropout_p, periodic=periodic) for _ in range(1, length=num_residual)],
+        
+        # Decoding
+        gnorm4=GroupNorm(channels[4], 32, swish),
+        tconv4=Upsampling(channels[4] => channels[3], nspatial, kernel_size=inner_kernelsize, periodic=periodic),
+        denset4=Dense(embed_dim, channels[3]),
+        tgnorm4=GroupNorm(channels[3], 32, swish),
+        
+        tconv3=Upsampling(channels[3]+channels[3] => channels[2], nspatial, kernel_size=middle_kernelsize, periodic=periodic),
+        denset3=Dense(embed_dim, channels[2]),
+        tgnorm3=GroupNorm(channels[2], 32, swish),
+        
+        tconv2=Upsampling(channels[2]+channels[2] => channels[1], nspatial, kernel_size=outer_kernelsize, periodic=periodic),
+        denset2=Dense(embed_dim, channels[1]),
+        tgnorm2=GroupNorm(channels[1], 32, swish),
+        
+        # Projection
+        tconv1=tconv1,
+        mean_bypass_layers...
+    )
+    
+    return NoiseConditionalScoreNetwork3D(layers, context)
+end
+
+@functor NoiseConditionalScoreNetwork3D
+
+"""
+    (net::NoiseConditionalScoreNetwork3D)(x, c, t)
+
+Evaluates the neural network of the NoiseConditionalScoreNetwork
+model on (x,c,t), where `x` is the tensor of noised input,
+`c` is the tensor of contextual input, and `t` is a tensor of times.
+"""
+function (net::NoiseConditionalScoreNetwork3D)(x, c, t)
+    # Get size of spatial dimensions
+    nspatial = ndims(x) - 2
+
+    # Embedding
+    embed = net.layers.gaussfourierproj(t)
+    embed = net.layers.linear(embed)
+
+    # remove mean of noised variables before input
+    h1 = x .- mean(x, dims=(1:nspatial)) 
+
+    # Encoder
+    h1 = concatenate_channels(Val(net.context), h1, c, nspatial)
+    h1 = net.layers.conv1(h1)
+    h1 = h1 .+ expand_dims(net.layers.dense1(embed), nspatial)
```

Comment:
here we go from something with only a channel dimension to something with nspatial+1 dimensions. then broadcast over batch?